### DISCUSSION TITLE
Keep the cursor of the current position always in the center

### DISCUSSION THREAD

**Discussion by Lukaszm328** [katspaugh/wavesurfer.js#3967] [Q&A] [answered]
I have a question about how to keep the cursor of the current position always in the center.
When i looping with regions or seek position.

this is very useful. I can manipulate the setScroll function but it is not a completely good solution. 

when loading a song, the cursor moves from the left. I need there to be nothing on the left side, just the cursor to start in the middle and a path to run from the middle.

wrong 
![image](https://github.com/user-attachments/assets/b9c2d5ab-4f6e-4803-8bff-858185a32c93)

i need like 
![image](https://github.com/user-attachments/assets/cee1fc4a-0258-4584-a7a4-321268ebb265)

**Comment by katspaugh** [comment]
Auto-centering is on by default but only when it reaches the center, see here: https://wavesurfer.xyz/examples/?zoom.js
There's no option to scroll emptiness to the middle.

**Reply by Lukaszm328** [reply]
maybe it is possible to add such functionality since we provide the auto center option?

**Reply by katspaugh** [reply, marked as answer]
I don't have time atm but happy to review a PR.

**Reply by Lukaszm328** [reply]
can you promise me that this functionality will be added? otherwise I will have to change the waveform library in my project.
